core/services/workflows/syncer/backoff.go

```diff
@@ -0,0 +1,105 @@
+package syncer
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+type backoffRecord struct {
```

Comment:
Did you explore what's already available to avoid reimplementing standard stuff? For example I see we have core/utils/backoff_ticker.go - maybe that could be useful here?

Reply:
I did -- it's not really a good fit here since the reconcile loop is constantly generating a diff between the actual and expected state; it's not easy to use a ticker to modulate when those events fire.

We could move the backoff logic to the feeding routine but the problem with that is that we then won't know whether the call to the handler has succeeded or not